### PR TITLE
Add and change search functions

### PR DESCRIPTION
**Pull request by iattempt**
**Fixes issue:**

**Changes:**

1. fix coding style
2. rename file
3. fix import location
4. change algorithm

### CODE REVIEW

File: code/search/src/exponential_search/exponential_search.py [hunk 1]
```diff
@@ -33,4 +33,4 @@ def exponential_search(arr, element):
     while i < arrLength and arr[i] <= element:
         i *= 2
 
-    return _binary_search(arr, elment, i / 2, min(i, arrLength))
```
Comment (iattempt) [inline code/search/src/exponential_search/exponential_search.py]: fix typo

File: code/search/src/binary_search/binary_search.py [hunk 2]
```diff
@@ -3,26 +3,29 @@
...
+def _binary_search_recursive_impl(arr, x, l, r):
+    if r > l:
         mid = l + int((r - l) / 2)
 
         if arr[mid] == x:
             return mid
 
         elif arr[mid] > x:
-            return binarySearchRecursive(arr, l, mid - 1, x)
+            return _binary_search_recursive_impl(arr, x, l, mid - 1)
```
Comment (arnavb) [inline code/search/src/binary_search/binary_search.py]: According to the [PEP 8](https://www.python.org/dev/peps/pep-0008/) style guide, I don't think you should be using a leading underscore for the name.

File: code/search/src/binary_search/binary_search.py [hunk 3]
```diff
@@ -32,13 +35,11 @@ def binarySearchLoop(arr, l, r, x):
         else:
             l = mid + 1
 
-    return -1
+    return l if l >= 0 and l < len(arr) and arr[l] == x else -1
```
Comment (arnavb) [inline code/search/src/binary_search/binary_search.py]:
```Python
return l if l in range(0, len(arr)) # ...
```

File: code/search/src/exponential_search/exponential_search.py [hunk 4]
```diff
@@ -0,0 +1,34 @@
...
+        if arr[mid] == element:
+            return mid
+
+        elif arr[mid] > element:
+            return _binary_search(arr, element, left, mid - 1)
+
+        else:
+            return _binary_search(arr, element, mid + 1, right)
+
+    return left if left >= 0 and left < len(arr) and arr[left] == element else -1
```
Comment (arnavb) [inline code/search/src/exponential_search/exponential_search.py]: Same here

File: code/search/src/binary_search/binary_search.py (hunk 2 above)
Comment (iattempt) [reply code/search/src/binary_search/binary_search.py]: I think the method is private and [PEP 8 - Method names "Use one leading underscore only for non-public methods"](https://www.python.org/dev/peps/pep-0008/#method-names-and-instance-variables), thus I added leading underscore.

**Comment by iattempt** [comment]
@arnavb Fixed, please review 😊.

**Comment by AdiChat** [comment]
Looks great 😍 

Hope you are enjoying your journey 🚆  with **OpenGenus**

**Pull request merged**